File: Fray.Command/InvokeFrayCommand.fs
namespace Fray

open System
open System.Management.Automation

[<Cmdlet(VerbsLifecycle.Invoke, "Fray")>]
[<OutputType(typeof<string>)>]
type InvokeFrayCommand() =
    inherit PSCmdlet()

    [<Parameter(ValueFromPipeline = true, ValueFromPipelineByPropertyName = true)>]
    member val InputObject: obj = "" with get, set

    [<Parameter>]
    member val TargetCase: Case = Case.Kebab with get, set

    override this.BeginProcessing () : unit = base.BeginProcessing ()

    override this.ProcessRecord () : unit =

        let strVal =
            match this.InputObject with
            | :? string as s -> s
            | o -> o.ToString ()

        let result = Fray.Invoke (strVal, this.TargetCase)

        this.WriteObject result

    override this.EndProcessing () : unit = base.EndProcessing ()

File: Fray.Command/InvokeFrayCommand.cs
using System.Management.Automation;

namespace Fray;

[Cmdlet(VerbsLifecycle.Invoke, "Fray")]
[OutputType(typeof(string))]
public class InvokeFrayCommand : PSCmdlet
{

	[Parameter(
		ValueFromPipeline = true,
		ValueFromPipelineByPropertyName = true,
		Position = 0)]
	public string InputObject { get; set; } = string.Empty;

	[Parameter]
	[Alias("Case")]
	public Case TargetCase { get; set; } = Case.Kebab;

	protected override void ProcessRecord()
	{
		try
		{
			string result = Fray.Invoke(InputObject, TargetCase);

			WriteObject(result);
		}
		catch (Exception ex)
		{
			WriteError(new ErrorRecord(
				ex,
				"FrayTransformError",
				ErrorCategory.NotSpecified,
				InputObject));
		}
	}
}
